### PR TITLE
CLI optimize database

### PR DESCRIPTION
**Pull request by Alkarex**
https://github.com/FreshRSS/FreshRSS/issues/1583
And VACUUM in SQLite https://github.com/FreshRSS/FreshRSS/issues/918

### CODE REVIEW

File: cli/db-optimize.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/php
+<?php
+require('_cli.php');
+
+$options = getopt('', array(
+		'user:',
+	));
+
+if (empty($options['user'])) {
+	fail('Usage: ' . basename(__FILE__) . " --user username");
```
Comment (aledeg) [inline cli/db-optimize.php]: Can't you just use ```__DIR__``` instead of ```basename(__FILE__)```?

**Review by aledeg — APPROVED** [review]
I've tested for MySql. It's working fine. Good job. Should I move the i18n tools in the cli folder?

**Comment by Alkarex** [comment]
Ah yes, this sounds like a good idea. I have not looked much at your i18n tools yet, but I assume they are all invoked from command line, aren't they?

**Comment by Alkarex** [comment]
Ah, PostgreSQL also has `VACUUM` https://www.postgresql.org/docs/9.2/static/sql-vacuum.html

**Comment by Alkarex** [comment]
Would some our of dear PostgreSQL users be willing to give a try to our VACUUM implementation? :-)
@anayrat @Trim @tcitworld ?
It is available in the dev branch of FreshRSS, accessible by CLI or with a Web button in FreshRSS settings.

**Comment by anayrat** [comment]
Hello,
I don't do tests for the moment. But normally, manual vacuum is not necessary. PostgreSQL perform autovacuum (enabled by default) depending on the activity (row inserted, updated...) : https://www.postgresql.org/docs/current/static/routine-vacuuming.html#autovacuum

Eventually, you can perform `VACUUM FULL` which rewrite all the tables and rebuild indexes. But it acquire an exclusive lock. It must be an exceptional maintenance.

**Pull request merged**